nifi-nar-bundles/nifi-adx-bundle/nifi-adx-processors/src/main/java/org/apache/nifi/processors/adx/AzureAdxIngestProcessor.java

```diff
@@ -0,0 +1,437 @@
+
+            switch (context.getProperty(IR_METHOD).getValue()) {
+                case "IRM_TABLE": ingestionProperties.setReportMethod(IngestionProperties.IngestionReportMethod.TABLE); break;
+                case "IRM_QUEUE": ingestionProperties.setReportMethod(IngestionProperties.IngestionReportMethod.QUEUE); break;
+                case "IRM_TABLEANDQUEUE": ingestionProperties.setReportMethod(IngestionProperties.IngestionReportMethod.QUEUE_AND_TABLE); break;
+            }
+
+            if (StringUtils.equalsIgnoreCase(context.getProperty(FLUSH_IMMEDIATE).getValue(),"true")) {
+                ingestionProperties.setFlushImmediately(true);
+            } else {
+                ingestionProperties.setFlushImmediately(false);
+            }
+
+            getLogger().info("Ingesting with: "
+                    + "dataFormat - "+ ingestionProperties.getDataFormat() + "|"
+                    + "ingestionMapping - " + ingestionProperties.getIngestionMapping().getIngestionMappingReference() + "|"
+                    + "reportLevel - " + ingestionProperties.getReportLevel() + "|"
+                    + "reportMethod - " + ingestionProperties.getReportMethod() + "|"
+                    + "databaseName - " + ingestionProperties.getDatabaseName() + "|"
+                    + "tableName - " + ingestionProperties.getTableName() + "|"
+                    + "flushImmediately - " + ingestionProperties.getFlushImmediately()
+            );
+
+            StreamSourceInfo info = new StreamSourceInfo(in);
+
+            IngestionResult result = client.ingestFromStream(info, ingestionProperties);
+
+            List<IngestionStatus> statuses = result.getIngestionStatusCollection();
+
+            if(StringUtils.equalsIgnoreCase(context.getProperty(WAIT_FOR_STATUS).getValue(),"ST_SUCCESS")) {
+                while (statuses.get(0).status == OperationStatus.Pending) {
+                    Thread.sleep(50);
+                    statuses = result.getIngestionStatusCollection();
+                    if(statuses.get(0).status == OperationStatus.Succeeded) {
+                        break;
+                    } else if (statuses.get(0).status == OperationStatus.Failed) {
+                        break;
+                    }
+                }
+            } else {
+                IngestionStatus status = new IngestionStatus();
+                status.status = OperationStatus.Succeeded;
+                statuses.set(0, status);
+            }
+
+            getLogger().info("Operation status: " + statuses.get(0).details);
+
+            if(statuses.get(0).status == OperationStatus.Succeeded) {
+                getLogger().info(statuses.get(0).status.toString());
+                session.transfer(flowFile, RL_SUCCEEDED);
+            }
+
+            if(statuses.get(0).status == OperationStatus.Failed) {
+                getLogger().error(statuses.get(0).status.toString());
+                session.transfer(flowFile, RL_FAILED);
+            }
```

Comment:
How are transactions handled ? What happens if we stream 100 records and there is failure after 99 rows ? Remember that the sink is not relational and this will cause duplicates , please check other sinks like AzureBlob / S3 or something similar